src/main/java/edu/ucsb/cs56/ucsb_open_lab_scheduler/services/CSVToObjectService.java

```diff
@@ -8,17 +8,19 @@
 import java.io.InputStreamReader;
 import java.io.Reader;
 import java.util.List;
-import java.util.Objects;
 
 
-public interface CSVToObjectService {
-    <T extends Object> List<T> parse(Reader csv);
+public interface CSVToObjectService<T> {
+    List<T> parse(Reader csv, Class<T> type);
 
     Logger getLogger();
 
-    default <T extends Object> List<T> parse(MultipartFile file) {
+    default List<T> parse(MultipartFile file, Class<T> type) {
         try {
-            return parse(new InputStreamReader(file.getInputStream()));
+            InputStreamReader reader = new InputStreamReader(file.getInputStream());
+            List<T> result = parse(reader, type);
+            reader.close();
```

Comment:
This does not close the reader in all cases. The close statement must be put in a finally block or preferably, a [try-with-resources block](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) should be used when creating the reader.

Reply:
Thanks! Updated.